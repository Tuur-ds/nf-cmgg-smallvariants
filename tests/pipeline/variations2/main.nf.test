nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_variations2"

    options "-stub"

    test("pipeline_variations - normalize") {

        when {
            params {
                outdir = "${outputDir}"
                normalize = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert !workflow.stdout },
                { assert snapshot(
                    path("${outputDir}")
                        .list()
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                        .findAll {
                            !(it.contains("/execution_") || it.contains("/params_") || it.contains("/pipeline_"))
                        }
                        .collect {
                            it.replace(getDynamicOutputName(), "<version>_<date>")
                        }
                ).match() }
            )
        }

    }

    test("pipeline_variations - updio") {

        when {
            params {
                outdir = "${outputDir}"
                updio = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert !workflow.stdout },
                { assert snapshot(
                    path("${outputDir}")
                        .list()
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                        .findAll {
                            !(it.contains("/execution_") || it.contains("/params_") || it.contains("/pipeline_"))
                        }
                        .collect {
                            it.replace(getDynamicOutputName(), "<version>_<date>")
                        }
                ).match() }
            )
        }

    }

    test("pipeline_variations - gemini") {

        when {
            params {
                outdir = "${outputDir}"
                gemini = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert !workflow.stdout },
                { assert snapshot(
                    path("${outputDir}")
                        .list()
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                        .findAll {
                            !(it.contains("/execution_") || it.contains("/params_") || it.contains("/pipeline_"))
                        }
                        .collect {
                            it.replace(getDynamicOutputName(), "<version>_<date>")
                        }
                ).match() }
            )
        }

    }

    test("pipeline_variations - validate") {

        when {
            params {
                outdir = "${outputDir}"
                validate = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert !workflow.stdout },
                { assert snapshot(
                    path("${outputDir}")
                        .list()
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                        .findAll {
                            !(it.contains("/execution_") || it.contains("/params_") || it.contains("/pipeline_"))
                        }
                        .collect {
                            it.replace(getDynamicOutputName(), "<version>_<date>")
                        }
                ).match() }
            )
        }

    }

    test("pipeline_variations - add_ped") {

        when {
            params {
                outdir = "${outputDir}"
                add_ped = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert !workflow.stdout },
                { assert snapshot(
                    path("${outputDir}")
                        .list()
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                        .findAll {
                            !(it.contains("/execution_") || it.contains("/params_") || it.contains("/pipeline_"))
                        }
                        .collect {
                            it.replace(getDynamicOutputName(), "<version>_<date>")
                        }
                ).match() }
            )
        }

    }

}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}

def getDynamicOutputName() {
    def Map nfcoreYaml = new groovy.yaml.YamlSlurper().parseText(file(".nf-core.yml").text)
    def date = new java.text.SimpleDateFormat("yyyy_MM_dd").format(new Date())
    return "v${nfcoreYaml.template.version.replace('.', '_')}_${date}" as String
}