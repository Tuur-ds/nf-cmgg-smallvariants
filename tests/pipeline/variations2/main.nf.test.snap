{
    "gemini": {
        "content": [
            [
                
            ],
            [
                "NA24143/NA24143.bed",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "NA24143/reports/NA24143.haplotypecaller.bcftools_stats.txt",
                "NA24149/NA24149.bed",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "NA24149/reports/NA24149.haplotypecaller.bcftools_stats.txt",
                "NA24385/NA24385.bed",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "NA24385/reports/NA24385.haplotypecaller.bcftools_stats.txt",
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.bed",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.db",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.ped",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:55:45.617425912"
    },
    "normalize": {
        "content": [
            [
                
            ],
            [
                "NA24143/NA24143.bed",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "NA24143/reports/NA24143.haplotypecaller.bcftools_stats.txt",
                "NA24149/NA24149.bed",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "NA24149/reports/NA24149.haplotypecaller.bcftools_stats.txt",
                "NA24385/NA24385.bed",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "NA24385/reports/NA24385.haplotypecaller.bcftools_stats.txt",
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.bed",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.ped",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:53:40.939481984"
    },
    "updio": {
        "content": [
            [
                
            ],
            [
                "NA24143/NA24143.bed",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "NA24143/reports/NA24143.haplotypecaller.bcftools_stats.txt",
                "NA24149/NA24149.bed",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "NA24149/reports/NA24149.haplotypecaller.bcftools_stats.txt",
                "NA24385/NA24385.bed",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "NA24385/reports/NA24385.haplotypecaller.bcftools_stats.txt",
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.bed",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.ped",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:54:44.363989965"
    },
    "add_ped": {
        "content": [
            [
                
            ],
            [
                "NA24143/NA24143.bed",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "NA24143/reports/NA24143.haplotypecaller.bcftools_stats.txt",
                "NA24149/NA24149.bed",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "NA24149/reports/NA24149.haplotypecaller.bcftools_stats.txt",
                "NA24385/NA24385.bed",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "NA24385/reports/NA24385.haplotypecaller.bcftools_stats.txt",
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.bed",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.ped",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:58:01.026718708"
    },
    "validate": {
        "content": [
            [
                
            ],
            [
                "NA24143/NA24143.bed",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "NA24143/reports/NA24143.haplotypecaller.bcftools_stats.txt",
                "NA24143/validation/haplotypecaller/NA24143.fn.vcf.gz",
                "NA24143/validation/haplotypecaller/NA24143.fn.vcf.gz.tbi",
                "NA24143/validation/haplotypecaller/NA24143.fp.vcf.gz",
                "NA24143/validation/haplotypecaller/NA24143.fp.vcf.gz.tbi",
                "NA24143/validation/haplotypecaller/NA24143.non_snp.png",
                "NA24143/validation/haplotypecaller/NA24143.non_snp.svg",
                "NA24143/validation/haplotypecaller/NA24143.non_snp_roc.tsv.gz",
                "NA24143/validation/haplotypecaller/NA24143.phasing.txt",
                "NA24143/validation/haplotypecaller/NA24143.snp.png",
                "NA24143/validation/haplotypecaller/NA24143.snp.svg",
                "NA24143/validation/haplotypecaller/NA24143.snp_roc.tsv.gz",
                "NA24143/validation/haplotypecaller/NA24143.summary.txt",
                "NA24143/validation/haplotypecaller/NA24143.tp-baseline.vcf.gz",
                "NA24143/validation/haplotypecaller/NA24143.tp-baseline.vcf.gz.tbi",
                "NA24143/validation/haplotypecaller/NA24143.tp.vcf.gz",
                "NA24143/validation/haplotypecaller/NA24143.tp.vcf.gz.tbi",
                "NA24143/validation/haplotypecaller/NA24143.weighted.png",
                "NA24143/validation/haplotypecaller/NA24143.weighted.svg",
                "NA24143/validation/haplotypecaller/NA24143.weighted_roc.tsv.gz",
                "NA24149/NA24149.bed",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "NA24149/reports/NA24149.haplotypecaller.bcftools_stats.txt",
                "NA24149/validation/haplotypecaller/NA24149.fn.vcf.gz",
                "NA24149/validation/haplotypecaller/NA24149.fn.vcf.gz.tbi",
                "NA24149/validation/haplotypecaller/NA24149.fp.vcf.gz",
                "NA24149/validation/haplotypecaller/NA24149.fp.vcf.gz.tbi",
                "NA24149/validation/haplotypecaller/NA24149.non_snp.png",
                "NA24149/validation/haplotypecaller/NA24149.non_snp.svg",
                "NA24149/validation/haplotypecaller/NA24149.non_snp_roc.tsv.gz",
                "NA24149/validation/haplotypecaller/NA24149.phasing.txt",
                "NA24149/validation/haplotypecaller/NA24149.snp.png",
                "NA24149/validation/haplotypecaller/NA24149.snp.svg",
                "NA24149/validation/haplotypecaller/NA24149.snp_roc.tsv.gz",
                "NA24149/validation/haplotypecaller/NA24149.summary.txt",
                "NA24149/validation/haplotypecaller/NA24149.tp-baseline.vcf.gz",
                "NA24149/validation/haplotypecaller/NA24149.tp-baseline.vcf.gz.tbi",
                "NA24149/validation/haplotypecaller/NA24149.tp.vcf.gz",
                "NA24149/validation/haplotypecaller/NA24149.tp.vcf.gz.tbi",
                "NA24149/validation/haplotypecaller/NA24149.weighted.png",
                "NA24149/validation/haplotypecaller/NA24149.weighted.svg",
                "NA24149/validation/haplotypecaller/NA24149.weighted_roc.tsv.gz",
                "NA24385/NA24385.bed",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "NA24385/reports/NA24385.haplotypecaller.bcftools_stats.txt",
                "NA24385/validation/haplotypecaller/NA24385.fn.vcf.gz",
                "NA24385/validation/haplotypecaller/NA24385.fn.vcf.gz.tbi",
                "NA24385/validation/haplotypecaller/NA24385.fp.vcf.gz",
                "NA24385/validation/haplotypecaller/NA24385.fp.vcf.gz.tbi",
                "NA24385/validation/haplotypecaller/NA24385.non_snp.png",
                "NA24385/validation/haplotypecaller/NA24385.non_snp.svg",
                "NA24385/validation/haplotypecaller/NA24385.non_snp_roc.tsv.gz",
                "NA24385/validation/haplotypecaller/NA24385.phasing.txt",
                "NA24385/validation/haplotypecaller/NA24385.snp.png",
                "NA24385/validation/haplotypecaller/NA24385.snp.svg",
                "NA24385/validation/haplotypecaller/NA24385.snp_roc.tsv.gz",
                "NA24385/validation/haplotypecaller/NA24385.summary.txt",
                "NA24385/validation/haplotypecaller/NA24385.tp-baseline.vcf.gz",
                "NA24385/validation/haplotypecaller/NA24385.tp-baseline.vcf.gz.tbi",
                "NA24385/validation/haplotypecaller/NA24385.tp.vcf.gz",
                "NA24385/validation/haplotypecaller/NA24385.tp.vcf.gz.tbi",
                "NA24385/validation/haplotypecaller/NA24385.weighted.png",
                "NA24385/validation/haplotypecaller/NA24385.weighted.svg",
                "NA24385/validation/haplotypecaller/NA24385.weighted_roc.tsv.gz",
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.bed",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.ped",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:56:56.038825116"
    }
}